### ISSUE TITLE
Assigning benedict element to itself clears the element

### ISSUE THREAD

**Issue by cdoolin** [fabiocaccamo/python-benedict#294]
**Python version**
3.11.10

**Package version**
0.30.0

**Current behavior (bug description)**
```python
d = benedict({
    "a": {
        "b": 23
    }
})
print(d)
# {'a': {'b': 23}} 

d.a = d.a
print(d)
# {'a': {}}
```

**Expected behavior**
Assigning `d.a = d.a` should not change the dictionary.  I would expect the following:

```python
d = benedict({
    "a": {
        "b": 23
    }
})
print(d)
# {'a': {'b': 23}} 

d.a = d.a
print(d)
# {'a': {'b': 23}} 
```

Cheers

<a href="https://dashboard.polar.sh/fabiocaccamo/python-benedict/issues/294">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://dashboard.polar.sh/api/github/fabiocaccamo/python-benedict/issues/294/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://dashboard.polar.sh/api/github/fabiocaccamo/python-benedict/issues/294/pledge.svg">
</picture>
</a>

**Comment by fabiocaccamo** [comment]
@cdoolin thank you for reporting this.

**Comment by fabiocaccamo** [comment]
Fixed in `0.30.2` version.
